lib/src/components/NxAccordion/__tests__/NxAccordion.test.tsx

```diff
@@ -5,313 +5,202 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 import React from 'react';
-import { mount } from 'enzyme';
-import { faChevronCircleDown, faChevronCircleUp } from '@fortawesome/free-solid-svg-icons';
-import { act } from 'react-dom/test-utils';
+import userEvent from '@testing-library/user-event';
 
-import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
-import NxAccordion, { Props } from '../NxAccordion';
-import NxButton from '../../NxButton/NxButton';
-import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
 
-function createClickEvent() {
-  return new MouseEvent('click', { bubbles: true });
-}
+import NxAccordion from '../NxAccordion';
+import NxButton from '../../NxButton/NxButton';
+import { render } from '@testing-library/react';
 
 describe('NxAccordion', function() {
-  const getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxAccordion, {}),
-      getMountedComponent = enzymeUtils.getMountedComponent<Props>(NxAccordion, {});
+  const quickRender = rtlRender(NxAccordion, {});
+  const renderEl = rtlRenderElement(NxAccordion, {});
 
   it('renders a <details> element with the provided props', function() {
-    const component = getShallowComponent({ id: 'foo', open: true, title: 'test title' });
-
-    expect(component).toMatchSelector('details#foo');
-    expect(component).toHaveProp('open', true);
-    expect(component).toHaveProp('title', 'test title');
+    const el = renderEl({ id: 'foo', open: true })!;
+    expect(el.tagName).toBe('DETAILS');
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('open');
   });
 
-  it('sets the nx-accordion class along with any provided className', function() {
-    expect(getShallowComponent()).toHaveClassName('nx-accordion');
-    expect(getShallowComponent({ className: 'foo' })).toHaveClassName('nx-accordion');
-    expect(getShallowComponent({ className: 'foo' })).toHaveClassName('foo');
+  it('sets the provided className', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
   it('sets an id if none is specified', function() {
```

Comment:
Rather than this test, I think what we really need is additional testing of the `aria-controls` on the header, for the case where an explicit id was not used.

Reply:
> 

Added it here: 7bd9b2c0b3e19a89c13646166fef04205e303e4a

Reply:
> Added it here: https://github.com/sonatype/sonatype-react-shared-components/commit/7bd9b2c0b3e19a89c13646166fef04205e303e4a

The ids are randomly/dynamically generated.  Hardcoding the value it might be is definitely not going to work.  The test should just test that the value of `aria-controls` and the `id` of the accordion match and are truthy, not that they are any specific value.

Reply:
> > Added it here: [7bd9b2c](https://github.com/sonatype/sonatype-react-shared-components/commit/7bd9b2c0b3e19a89c13646166fef04205e303e4a)
> 
> The ids are randomly/dynamically generated. Hardcoding the value it might be is definitely not going to work. The test should just test that the value of `aria-controls` and the `id` of the accordion match and are truthy, not that they are any specific value.

Updated: 91d01901daf3ebdebbac2d98fb6a2d7c5cf78b5f